hyper/http11/response.py

```diff
@@ -48,7 +49,8 @@ def __init__(self, code, reason, headers, sock, connection=None):
         # bother checking for content-length, we just keep reading until
         # we no longer can.
         self._expect_close = False
-        if b'close' in self.headers.get(b'connection', []):
+        if (b'close' in self.headers.get(b'connection', []) or
+                b'close' in self.headers.get(b'proxy-connection', [])):
             self._expect_close = True
```

Comment:
Yes, sure, but this is not a response to CONNECT.

Let me give you a bigger picture on this.
Imagine that we totally ignore the `Proxy-Connection` header.
Consider this situation:

- HTTP/1.1 connection,
- plaintext (`secure=False`),
- with proxy (that is a simple proxying without CONNECT tunnel),
- which requires authentication.

Say we have sent this request to the proxy:

	GET http://stackoverflow.com/ HTTP/1.1
	User-Agent: python-requests/2.13.0
	Accept-Encoding: gzip, deflate
	Accept: */*
	Connection: keep-alive
	proxy-authorization: Basic aGk6aGkK
	connection: Upgrade, HTTP2-Settings
	upgrade: h2c
	HTTP2-Settings: AAQAAP__AAIAAAAA
	host: stackoverflow.com

Proxy might respond with a following:

	HTTP/1.0 407 Proxy Authentication Required
	Proxy-Authenticate: Basic realm="proxy"
	Proxy-Connection: close
	Content-type: text/html; charset=utf-8

	<html><head><title>407 Proxy Authentication Required</title></head>
	<body><h2>407 Proxy Authentication Required</h2><h3>Access to requested resource disallowed by administrator or you need valid username/password to use this resource</h3></body></html>

Note that:
- this is not a response to CONNECT,
- there's no `connection: close` header,
- there's no `transfer-encoding: chunked` header,
- there's no `content-length` header,
- but there's a body, thus I suggest that the `Proxy-Connection: close` header is used akin the `connection: close` header here.

Do you think that there might be a situation when treating the `Proxy-Connection: close` header as `connection: close` would cause unintended effects?